webapp/niapp.yaml

```diff
@@ -0,0 +1,416 @@
+    schema:
+      type: object
+      title: Update WebApp Sharing Request
+      properties:
+        shared:
+          type: string
+          enum:
+            - private
+            - direct
+            - public
+          description: The webapp's sharing option
+        sharedEmails:
+          type: array
+          items:
+            type: string
+          description: List of emails of users to share the webapp with. Applies when "shared" option is "direct"
+
+responses:
+  Unauthorized:
+    description: API Key is missing or invalid
+  NotFound:
+    description: The resource was not found.
+  ValidationError:
+    description: Invalid input data
+  GetWebAppResponse:
+    description: Get WebApp Response
+    schema:
+      $ref: '#/definitions/WebApp'
+  CreateWebAppResponse:
+    description: Create WebApp Response
+    schema:
+      $ref: '#/definitions/WebApp'
+  UpdateWebAppResponse:
+    description: Update WebApp Response
+    schema:
+      $ref: '#/definitions/WebApp'
+  UpdateWebAppSharingResponse:
+    description: Update WebApp Sharing Response
+    schema:
+      $ref: '#/definitions/WebApp'
+  ListWebAppsResponse:
+    description: List WebApps Response
+    schema:
+      title: List WebApps Response
+      type: object
+      properties:
+        webapps:
+          description: List of webapps
+          type: array
+          items:
+            $ref: '#/definitions/WebApp'
+        continuationToken:
+          description: The continuation token can be used to paginate through the webapp list results. Provide this token in the next list webapps call.
+          type: string
+  SharedEmailsResponse:
+    description: Shared Emails Response
+    schema:
+      title: Shared Emails Response
+      type: array
+      items:
+        type: string
+        example: john.doe@email.com
+
+definitions:
+  WebApp:
+    type: object
+    title: WebApp Metadata
+    properties:
+      id :
+        type: string
+        description: The webapp id
+        example: "asdsad-17a6-45323-b64b-65325287372d"
+      type:
+        type: string
+        enum:
+          - Dashboard
+          - DashboardTemplate
+          - WebVI
+        description: The webapp type
+      name:
+        type: string
+        description: The webapp name
+      shared:
+        type: string
+        enum:
+          - private
+          - direct
+          - public
+        description: The webapp's sharing option
+      sharedEmails:
+        type: array
+        items:
+          type: string
+          example: john.doe@ni.com
+        description: List of emails of users to share the webapp with. Applies when "shared" option is "direct"
+      policyIds:
+        type: array
+        items:
+          type: string
+          example: "asdsad-17a6-45323-b64b-65325287372d"
+        description: List of policy Ids associated with the webapp, which give it access to the user's resources"
+      created:
```

Comment:
OpenAPI supports an additional format specifier for timestamps stored as strings, `format: date-time`. Without this present, some code generators may not translate this field to a language-native datetime.